coa/pkg/apis/v1alpha2/events.go

```diff
@@ -23,6 +23,28 @@ func (e Event) MarshalBinary() (data []byte, err error) {
 
 type EventHandler func(topic string, message Event) error
 
+func EventShouldRetryWrapper(handler EventHandler, topic string, message Event) bool {
+	err := handler(topic, message)
+	if err != nil {
+		coaE, ok := err.(COAError)
+		if !ok {
+			// error is not a COAError, which comes from external system
+			// always retry
+			return true
+		}
+		switch coaE.State {
+		case BadRequest, Unauthorized, NotFound, BadConfig, MethodNotAllowed, Conflict, MissingConfig, InvalidArgument, DeserializeError, SerializationError:
```

Comment:
Extend COAError a function of isRetriableError. We might use this function in other places.